bin/inference/pycbc_inference_plot_samples

```diff
@@ -44,44 +52,87 @@ opts = parser.parse_args()
 # setup log
 pycbc.init_logging(opts.verbose)
 
+rc('text', usetex=True)
+
 # load the results
 fp, parameters, labels, _ = io.results_from_cli(opts,
-                                                load_samples=False)
+                                                load_samples=False,
+                                                walkers=None)
 
 # get number of dimensions
 ndim = len(parameters)
 
+# get walker indices
+if opts.walkers == ['all'] or opts.walkers == None:
+    walkers = range(fp.nwalkers)
+else:
+    walkers = list(map(int, opts.walkers))
+
 # plot samples
 # plot each parameter as a different subplot
 logging.info("Plotting samples")
-fig, axs = plt.subplots(ndim, sharex=True)
+fig, axs = plt.subplots(ndim, figsize=(10,8), sharex=True)
 plt.xlabel("Iteration")
 
 # loop over parameters
 axs = [axs] if not hasattr(axs, "__iter__") else axs
+
+file_parameters, cs = transforms.get_common_cbc_transforms(
+                                            parameters, fp.variable_params)
+if opts.thin_start is not None:
+    xmin = opts.thin_start
+else:
+    xmin = fp.attrs['thin_start']
+
+if opts.thin_end is not None:
+    xmax = opts.thin_end
+else:
+    try:
+        xmax = fp.attrs['thin_end']
+    except KeyError:
+        xmax = fp.niterations
+
+if opts.thin_interval is not None:
+    xint = opts.thin_interval
+else:
+    xint = fp.attrs['thin_interval']
+
+if xmin > xmax:
+    raise ValueError("Start iteration {} provided for reading samples is "
+                     "higher than the end iteration {}.".format(xmin, xmax))
+xvals = numpy.arange(xmin, xmax, xint)
```

Comment:
This won't work if the samples have been pre-thinned in the file using `max-samples-per-chain`. The issue is that the `thin_start`, `thin_end`, and `thin_interval` attributes refer to sample indices on disk, which not the same as the iteration count when pre-thinning is on. The conversion between sample index and iteration count is given by the `thinned_by` attribute.

The only thing you really need is the `thin_start` and the `thinned_by` attributes. So, I'd remove everything about thin end and thin interval. Instead, just get the `thinned_by` attribute. This is a property of MCMC hdf files; see https://github.com/gwastro/pycbc/blob/master/pycbc/inference/io/base_mcmc.py#L95, so you can just do:
```
thinned_by = fp.thinned_by
```
Then just create the xvalues on the fly when creating the plot (see my comment below).